ddtrace/settings.py

```diff
@@ -69,11 +70,109 @@ def _add(self, integration, settings, merge=True):
             # >>> config._add('requests', dict(split_by_domain=False))
             # >>> config.requests['split_by_domain']
             # True
-            self._config[integration] = deepmerge(existing, settings)
+            self._config[integration] = IntegrationConfig(self, deepmerge(existing, settings))
         else:
-            self._config[integration] = settings
+            self._config[integration] = IntegrationConfig(self, settings)
+
+    def trace_headers(self, whitelist):
+        """
+        Registers a set of headers to be traced at global level or integration level.
+        :param whitelist: the case-insensitive list of traced headers
+        :type whitelist: list of str or str
+        :return: self
+        :rtype: HttpConfig
+        """
+        self._http.trace_headers(whitelist)
+        return self
+
+    def header_is_traced(self, header_name):
+        """
+        Returns whether or not the current header should be traced.
+        :param header_name: the header name
+        :type header_name: str
+        :rtype: bool
+        """
+        return self._http.header_is_traced(header_name)
 
     def __repr__(self):
         cls = self.__class__
         integrations = ', '.join(self._config.keys())
         return '{}.{}({})'.format(cls.__module__, cls.__name__, integrations)
+
+
+class IntegrationConfig(dict):
+
+    def __init__(self, global_config, *args, **kwargs):
+        """
+        :param global_config:
+        :type global_config: Config
+        :param args:
+        :param kwargs:
+        """
+        super(IntegrationConfig, self).__init__(*args, **kwargs)
+        self.http = HttpConfig()
+        self.global_config = global_config
+
+    def __deepcopy__(self, memodict=None):
+        new = IntegrationConfig(self.global_config, deepcopy(dict(self)))
+        new.http = deepcopy(self.http)
+        return new
+
+    def header_is_traced(self, header_name):
+        """
+        Returns whether or not the current header should be traced.
+        :param header_name: the header name
+        :type header_name: str
+        :rtype: bool
+        """
+        return self.http.header_is_traced(header_name) \
+            if self.http.is_header_tracing_configured \
+            else self.global_config.header_is_traced(header_name)
+
+
+class HttpConfig(object):
+    """
+    Configuration object that expose an API to set and retrieve both global and integration specific settings
+    related to the http context.
+    """
+
+    def __init__(self):
+        self._whitelist_headers = []
```

Comment:
can we make this a `set()` instead?

and below change to `self._whitelist_headers.add(normalized_header_name)`

keeps from duplicates ending up in the list, and doing `item in set()` is faster than `item in list()`